added some submenu to select time by hours, minutes, seconds

F3l1XQu1ll opened the pull request:
Please notice that this will only allow to set the interval in hours, minutes and seconds; because using the spin button is painfully slow and ineffective. This might be changed in the future to control at which time of the day the image is going to be displayed …

However, in this current implementation it does only change the value of the current spin button and nothing more.
Controlling the start time would require to change the UI in the case that the fit day option is enabled - at least in my opinion.

_issue #55_

maoschanz commented:
ah.... Glade...

i don't understand what changed by reading the diffs, could you post a screenshot please?

F3l1XQu1ll commented:
Should look like this:
![Bildschirmfoto von 2020-07-22 19-55-39](https://user-images.githubusercontent.com/39009640/88211191-a45bfd80-cc55-11ea-9054-baf0b1b1c1f5.png)

maoschanz commented:
Ok i see. Could you please:

- put the code of this new popover in a distinct file
- duplicate its content (at the left, "Start time" label + 3 spin-buttons, at the right "End time" label + 3 spin-buttons, maybe a vertical gtk separator between the 2 halfs)
- rollback `picture_widget.ui`, where 90% of the changes are just glade being hostile to humans

----

Only do this first part, because for the following idea, i'm not sure how it'll look so i want to try various designs

----

As you said, the feature makes sense if and only if the wallpaper is designed to fit 24 hours. But in this case, the values to fill these spinbuttons are already computed somewhere to be shown in labels, whose visibility is already linked to whether or not it's a 24h-wallpaper

So once your popover is more reusable, the fix would "just" be to change the labels highlighted here

![Capture d’écran de 2020-07-22 20-07-32](https://user-images.githubusercontent.com/19996113/88212348-68299c80-cc57-11ea-8ab6-ffd1a21ed2d1.png)

into menubuttons *(for both the times where it's static and the times of the transition)*, where the relief is none, the popover is your file, and the label is the same as the current label.

(And then connect the spin-buttons to some methods, but i'll see this even later)

F3l1XQu1ll commented:
Sure, but won't be able to do this until tomorrow evening :thinking:

maoschanz commented:
Don't worry, take as much time as you want

F3l1XQu1ll commented:
Do you want me to just move the code to a new file or shall it be connected with the rest of the code?

F3l1XQu1ll commented:
Ok, I implemented some new class for the popover and created a new ui file as well as I reverted the old ui to the master state.
The new stuff is not connected to the main app and won't show up …
Here is a screenshot of the popup in glade:
![Bildschirmfoto von 2020-07-22 23-34-58](https://user-images.githubusercontent.com/39009640/88231152-fd3a8e80-cc73-11ea-9938-efea8bd6d668.png)

maoschanz commented:
perfect, i'll try to connect it to the rest, maybe this week
